frontend/src/metabase-lib/join.ts

```diff
@@ -98,3 +98,12 @@ export function joinFields(join: Join): ColumnMetadata[] {
 export function withJoinFields(join: Join, newFields: ColumnMetadata[]): Join {
   return ML.with_join_fields(join, newFields);
 }
+
+export function renameJoin(
+  query: Query,
+  stageIndex: number,
+  oldNameOrIndex: string | number,
```

Comment:
The target can be specified by its alias or index. Other option would be requiring the whole join clause. Input welcome!